google/resource_container_cluster.go

```diff
@@ -2680,11 +2688,14 @@ func flattenClusterAddonsConfig(c *containerBeta.AddonsConfig) []map[string]inte
 	}
 
 	if c.CloudRunConfig != nil {
-		result["cloudrun_config"] = []map[string]interface{}{
-			{
-				"disabled": c.CloudRunConfig.Disabled,
-			},
+		cloudRunConfig := map[string]interface{}{
+			"disabled": c.CloudRunConfig.Disabled,
+		}
+		if c.CloudRunConfig.LoadBalancerType == "LOAD_BALANCER_TYPE_INTERNAL" {
+			// Currently we only allow setting load_balancer_type to LOAD_BALANCER_TYPE_INTERNAL
+			cloudRunConfig["load_balancer_type"] = "LOAD_BALANCER_TYPE_INTERNAL"
```

Comment:
```suggestion
		cloudRunConfig := map[string]interface{}{
			"disabled":           c.CloudRunConfig.Disabled,
			"load_balancer_type": c.CloudRunConfig.LoadBalancerType,
```
now that we are passing the API value through, we don't need the if statement anymore

Reply:
I still want to keep the `if` check because the server side will return `LOAD_BALANCER_TYPE_EXTERNAL` for external case which we don't want to return and store into state file.

Reply:
Is it okay if we store it in state file? Terraform can handle this situation if we set the field schema as `Computed: true`
If the user doesn't explicitly state `load_balancer_type` and the API returns it, setting the field to computed will allow Terraform to default to the API value and not cause any diffs. 

Otherwise, we can keep the if and ignore `LOAD_BALANCER_TYPE_EXTERNAL`. I'm fine either way

Reply:
Thanks @c2thorn. I will just keep it as it is.